File: Searching/LinearSearching/LinearSearch.cpp
#include <iostream>

using namespace std;

int linearSearch(int arr[], int target, int size)
{
	for(int i = 0; i < size; i++)
	{
		if(arr[i] == target)
			return i;
	}
	return -1;
}


int main()
{
	int n, target;
	cout<<"Enter size of array ";
	cin>>n;
	int arr[n];
	cout<<"Enter "<<n<<" elements ";
	for(int i = 0; i < n; i++)
		cin>>arr[i];
	cout<<"What number are you looking for in the list? ";
	cin>>target;

	cout<<linearSearch(arr, target, n)<<endl;
}

File: Searching/LinearSearching/LinearSearch.java
import java.util.*;
public class LinearSearch{
	public static void main(String[] args)
	{
		Random r = new Random();
		LinearSearch linearSearch = new LinearSearch();
		int arr[] = new int[r.nextInt(20)];
		for(int j = 0; j < arr.length; j++){
			arr[j] = r.nextInt(2000);
		}
		int n = arr[r.nextInt(arr.length)];
		System.out.println(linearSearch.linearSearch(arr, n));
	}

	int linearSearch(int arr[], int n){
		for(int i = 0; i < arr.length; i++){
			if(arr[i] == n)
				return i;
		}
		return -1;
	}
}
